hack/create-identity-secret.sh

```diff
@@ -21,10 +21,13 @@ set +o xtrace
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 cd "${REPO_ROOT}" || exit 1
 
-# shellcheck source=hack/ensure-kubectl.sh
-source "${REPO_ROOT}/hack/ensure-kubectl.sh"
+# Installation of kubectl
+mkdir -p "${REPO_ROOT}/hack/tools/bin/kubectl"
+KUBECTL=$(realpath hack/tools/bin/kubectl)
+make "${KUBECTL}" &>/dev/null
```

Comment:
I don't think this is doing what is intended?  If I run these commands locally then try to run kubectl via the env variable I get:

```
$KUBECTL 
bash: /workspaces/cluster-api-provider-azure/hack/tools/bin/kubectl: Is a directory
```
It looks like this is only during the make create-management-cluster command. Have you tried running that?

https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/e5327290e4fd5df3fae5779296b8e1d2c73c508e/Makefile#L483-L491

Reply:
Line 25 should be `mkdir -p "${REPO_ROOT}/hack/tools/bin"` instead of `mkdir -p "${REPO_ROOT}/hack/tools/bin/kubectl"`

same everywhere else

Reply:
@CecileRobertMichon removing kubectl from the path gives `realpath: hack/tools/bin/kubectl: No such file or directory`